Box_intersection_d/include/CGAL/Box_intersection_d/segment_tree.h

```diff
@@ -275,9 +275,8 @@ split_points( RandomAccessIter begin, RandomAccessIter end,
 
 
 #if CGAL_BOX_INTERSECTION_DEBUG
- static int level = -1;
  #define CGAL_BOX_INTERSECTION_DUMP(msg) { \
-   for( unsigned int i = level; i; --i ) \
+   for( unsigned int i = -1; i; --i ) \
```

Comment:
I had not really understood the code. The variable `Counter<int> bla(level);`  increases level each time the function is called, so the for loop starts with `level == 0`.